crates/turbopack-ecmascript-plugins/src/transform/relay.rs

```diff
@@ -0,0 +1,72 @@
+use std::path::PathBuf;
+
+use async_trait::async_trait;
+use swc_core::{
+    common::{util::take::Take, FileName},
+    ecma::{
+        ast::{Module, Program},
+        visit::FoldWith,
+    },
+};
+use turbo_tasks_fs::FileSystem;
+use turbopack_ecmascript::{CustomTransformer, TransformContext};
+
+#[derive(Debug)]
+pub struct RelayTransformer {
+    config: swc_relay::Config,
+}
+
+impl RelayTransformer {
+    pub fn new(config: swc_relay::Config) -> Self {
+        Self { config }
+    }
+}
+
+#[async_trait]
+impl CustomTransformer for RelayTransformer {
+    async fn transform(
+        &self,
+        program: &mut Program,
+        ctx: &TransformContext<'_>,
+    ) -> Option<Program> {
+        let root = if let Ok(file_path) = ctx.file_path.await {
+            file_path
+                .fs
+                .root()
+                .await
+                .map_or(PathBuf::new(), |v| PathBuf::from(v.path.to_string()))
```

Comment:
That's always an empty PathBuf. The root of the fs has always `path = ""`

Reply:
Huh, didn't know. Is there a way to get an asbolute root in that case?

Reply:
if we can get a full absolute root, https://github.com/vercel/turbo/pull/4721#discussion_r1182870967 can be done as well.

Reply:
Updated https://github.com/vercel/turbo/pull/4721/files#diff-236047cd15903b7c05afa8b8857302452a4c02acb690a5ca98e5f372c4898aa3R32